graph/task.go

```diff
@@ -80,6 +80,16 @@ func UnmarshalTaskFromFile(file string, creds []*Credential) (*Task, error) {
 	return t, err
 }
 
+// NewTaskFromString unmarshals a Task from string without any initialization.
+func NewTaskFromString(data string) (*Task, error) {
+	t := &Task{}
```

Comment:
nit: a bit cleaner if you just return the error instead of wrapping it here; you end up handling it twice right now